ethers-providers/src/provider.rs

```diff
@@ -521,16 +521,15 @@ impl<P: JsonRpcClient> Middleware for Provider<P> {
 
     /// Signs typed data using a specific account (EIP712). This account needs to be unlocked.
     #[cfg(feature = "eip1193")]
-    async fn sign_typed_data<T: Into<Bytes> + Send + Sync>(
+    async fn sign_typed_data(
         &self,
-        data: T,
+        data: &str,
         from: &Address,
     ) -> Result<Signature, ProviderError> {
-        let data = utils::serialize(&data);
         let from = utils::serialize(from);
 
         // get the response from `eth_signTypedData_v4` call and trim the 0x-prefix if present.
-        let sig: String = self.request("eth_signTypedData_v4", [from, data]).await?;
+        let sig: String = self.request("eth_signTypedData_v4", (from, data)).await?;
```

Comment:
Nie do końca chciałbym chyba aż tak dziwnie, głównie dlatego, że w przypadku implementacji np dla wallet connect to niekoniecznie będzie string. Na razie zostawię to wiszące i muszę przejrzeć architekturę `ethers` żeby to wymyślić trochę lepiej